dao/src/test/java/com/epam/esm/hibernate/impl/CertificateRepositoryImplTest.java

```diff
@@ -0,0 +1,231 @@
+package com.epam.esm.hibernate.impl;
+
+import com.epam.esm.config.ConfigDB;
+import com.epam.esm.exception.DaoException;
+import com.epam.esm.hibernate.CertificateRepository;
+import com.epam.esm.persistence.GiftCertificateEntity;
+import com.epam.esm.persistence.TagEntity;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(
+        classes = {ConfigDB.class},
+        loader = AnnotationConfigContextLoader.class)
+class CertificateRepositoryImplTest {
```

Comment:
Common notice for all the tests:
- duplicated code
- strings, numbers should be moved to constants and reused